wry/wsmanResource.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python2
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import wsmanData
+import uuid
+import requests
+import WryDict
+from time import sleep
+
+'''
+Created on 10 Jul 2017
+
+@author: adrian
+'''
+
+class wsmanResource(object):
+    '''
+    Class to represent a resource on a wsman compatible server
+    '''
+
+    def __init__(self, target = None, is_ssl = False, username = None, password = None, resource = None, debug = False, showxml = False):
+        '''
+        Set up this resource
+
+        @param target: the hostname or IP address of the wsman service
+        @param is_ssl: should we communicate using SSL?
+        @param resource: the identifier of the resource containing the settings we are interested in
+        @param username: the username to log in with
+        @param password: the password to log in with
+        '''
+        if is_ssl:
+            scheme = 'https'
+        else:
+            scheme = 'http'
+        port = wsmanData.AMT_PROTOCOL_PORT_MAP[scheme]
+        self.target = scheme + "://" + target + ":" + str(port) + "/wsman"
+        self.resourceId = resource
+        self.resourceUri = wsmanData.RESOURCE_URIS[self.resourceId]
+        self.resource_methods = wsmanData.RESOURCE_METHODS[resource]
+        self.username = username
+        self.password = password
+        self.debug = debug
+        self.showxml = showxml
+
+    def etree_to_dict(self, t):
+        d = {t.tag : map(self.etree_to_dict, t.iterchildren())}
+        d.update(('@' + k, v) for k, v in t.attrib.iteritems())
+        d['text'] = t.text
+        return d
+
+    def request(self, doc = None, params = {}):
+        '''
+        Send a request to the target and return the response
+
+        '''
+        params['uuid'] = uuid.uuid4()
+        if enumerate:
+            doc = doc % params
+        else:
+            doc = doc % params
+        if self.showxml:
+            print "==== Request ===="
```

Comment:
prints?

Reply:
Debugging code to allow users of the library to dump the conversations. Enabled only if self.showxml is set